src/EntityComponentManager.cc

```diff
@@ -233,6 +253,24 @@ class ignition::gazebo::EntityComponentManagerPrivate
   /// \brief See above
   public: std::unordered_map<Entity, Entity> oldToClonedCanonicalLink;
 
+  /// \brief During cloning, we populate two maps:
+  ///  - map of link entities to their cloned link
+  ///  - map of cloned joint entities to the original joint entity's parent and
+  ///    child links
+  /// After cloning is done, these maps can be used to update the cloned joint
+  /// entity's parent and child links to the cloned parent and child links.
+  /// \TODO(anyone) We shouldn't be giving joints special treatment.
+  /// We should figure out a way to update a joint's parent/child links without
+  /// having to explicitly search/track for the cloned links.
```

Comment:
There's a similar `TODO` note here: https://github.com/ignitionrobotics/ign-gazebo/blob/8f083d6d212c5867cc18cd45162ba3e4807643c4/src/EntityComponentManager.cc#L248-L250

I don't think we have an issue for tracking the idea of "generalizable component cloning" yet, do we? Should we create one?